src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationProviderBase.cs

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    /// <summary>
+    /// Base class for worker configuration resolvers.
+    /// </summary>
+    internal abstract class WorkerConfigurationProviderBase : IWorkerConfigurationProvider
+    {
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<WorkerConfigurationResolverOptions> _resolverOptions;
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly IWorkerProfileManager _profileManager;
+        private readonly IFileSystem _fileSystem;
+        private readonly ISystemRuntimeInformation _systemRuntimeInformation;
+
+        public WorkerConfigurationProviderBase(ILoggerFactory loggerFactory,
+                                                    IMetricsLogger metricsLogger,
+                                                    IFileSystem fileSystem,
+                                                    IWorkerProfileManager workerProfileManager,
+                                                    ISystemRuntimeInformation systemRuntimeInformation,
+                                                    IOptionsMonitor<WorkerConfigurationResolverOptions> workerConfigurationResolverOptions)
+        {
+            ArgumentNullException.ThrowIfNull(loggerFactory);
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryWorkerConfig);
+            _resolverOptions = workerConfigurationResolverOptions ?? throw new ArgumentNullException(nameof(workerConfigurationResolverOptions));
+            _metricsLogger = metricsLogger ?? throw new ArgumentNullException(nameof(metricsLogger));
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _profileManager = workerProfileManager ?? throw new ArgumentNullException(nameof(workerProfileManager));
+            _systemRuntimeInformation = systemRuntimeInformation ?? throw new ArgumentNullException(nameof(systemRuntimeInformation));
+            ArgumentNullException.ThrowIfNull(_resolverOptions.CurrentValue);
+        }
+
+        protected ILogger Logger => _logger;
+
+        protected WorkerConfigurationResolverOptions WorkerResolverOptions => _resolverOptions.CurrentValue;
+
+        protected IMetricsLogger MetricsLogger => _metricsLogger;
+
+        protected IWorkerProfileManager ProfileManager => _profileManager;
+
+        protected IFileSystem FileSystem => _fileSystem;
```

Comment:
The base class should have properties only it needs directly for its own implementation. There is no usage of `IFileSystem` in this class so it should be removed.